packages/compiler-vapor/src/transforms/transformRef.ts

```diff
@@ -10,7 +11,10 @@ import { findProp } from '../utils'
 import { EMPTY_EXPRESSION } from './utils'
 
 export const transformRef: NodeTransform = (node, context) => {
-  if (node.type !== NodeTypes.ELEMENT) return
+  // If the current node has conditional directives,
+  // it will result in duplicate registration of SET_REF operations.
+  if (node.type !== NodeTypes.ELEMENT || hasConditionalDirective(node)) return
+
```

Comment:
The same problem also applies to `v-for`, and the reason for the issue is more likely due to an incorrect order of Transform Preset. After I adjusted the order, the problem disappeared.

![image](https://github.com/vuejs/core-vapor/assets/19204772/329f61be-63db-43e1-af01-2a2181289873)

Reply:
Oh, that's correct, but some of the behaviors of v for+ref don't seem to align with vue core. Perhaps another PR needs to
 
align these behaviors.